cmd/build/v2/services.go

```diff
@@ -27,47 +28,46 @@ import (
 )
 
 // GetServicesToBuild returns the services it has to built because they are not already built
-func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model.Manifest, svcToDeploy []string) ([]string, error) {
-	buildManifest := manifest.Build
+func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model.Manifest, servicesToDeploy []string) ([]string, error) {
```

Comment:
I think we need a bigger change. I'll try to explain myself.

In the `deploy` command, you can specify a list of services to be deployed, for example you can do `okteto deploy api frontend`. If there is only a docker compose, we will deploy only those 2 services defined within the compose (and we need to build those images if they are not created)

The problem becomes trickier when there is an okteto manifest with some commands and also referencing a docker-compose. For example:

I have an okteto manifest like this:
```
build:
  frontend:
    context: frontend

deploy:
  commands:
    - name: Deploy Frontend
      command: helm upgrade --install frontend frontend/chart --set image=${OKTETO_BUILD_FRONTEND_IMAGE}
  compose: docker-compose.yml
```

and a docker compose like this:

```
services:
  mysql:
    image: 'mysql:8.0.28-oracle'
    ports:
    - '3306:3306'
  nginx:
    image: nginx:latest
    container_name: webserver
    restart: unless-stopped
    ports:
      - 80:80
      - 443:443
    volumes:
      - /home/ubuntu/rstudio_docker/nginx.conf:/etc/nginx/nginx.conf
      - /home/ubuntu/ssl:/ssl/

```

and I execute `okteto deploy mysql`, it will have to deploy (and build if needed) the service `mysql` defined in the docker compose but it will also execute the commands defined in the `deploy` section, so it will have to build the `frontend` image. In this case, it doesn't have to build `nginx` image because it is not one of the services to be deployed.

So, we need to bear in mind the images defined within the `build` section, because some of them might have to be deployed even if we have a list of `servicesToDeploy`.

So, at the end, I think the candidate images to build are the ones in the `servicesToDeploy` + images defined in the build section that don't refer to any service in any compose file.

Then, when you have that list of images, you have to check if they have to be built or not (they might already exist in the registry)

@jLopezbarb @pchico83 am I right?

Reply:
> I think the candidate images to build are the ones in the servicesToDeploy + images defined in the build section that don't refer to any service in any compose file

What if the deploy uses images built from the compose? For example the command could be `echo ${OKTETO_BUILD_MYSQL_IMAGE} ${OKTETO_BUILD_FRONTEND_IMAGE}`. Wouldn't we need to build all from the compose too?

I find the behavior of `okteto deploy service` weird. Why would all the deploy commands get ran if I'm only deploying one `service`?

Reply:
Yes, the functionality must be as it is for the rest of the commands that are not `okteto deploy`. Changing it on the build command affects `stack deploy`/`up` which have some implications such as the one commented by @ifbyol or stacks not being able to mount volume hosts.

When we load a `manifestV2` and it has an stack we move the `services.Build` sections to the manifest struct, that way we only have to control one build section instead of two. The problem comes on `okteto deploy` where we have to build the services that comes from a `Stack` and the services that comes from the `okteto manifestV2`. Maybe we can encapsulate [this part of the  inferring from stack](https://github.com/okteto/okteto/blob/3636552ee541450e82c1904ba58a097b839e75c6/pkg/model/manifest.go#L879) to know the services that have a build section and comes from a compose/stack and use that to create `imagesToBuild` on the deploy command. After knowing which images comes from the `okteto manifest` itself and which ones from the `stack` (the stack is stored as it is, so there are no changes after reading it) we can add the ones from that the user selected on compose + the ones that are on the manifest and we should only use that list to build and not to deploy. 

This PR is looping over the manifest v2 build section but since it has an `okteto stack` we already have those images(the ones that comes from the `octeto stack`) to build on the struct

Out of scope: We should return an error if the manifest on the deploy section has `svcToDeploy` but doesn't have a compose in it.

Reply:
> In another topic, I find the behavior of okteto deploy service weird. Why would all the deploy commands get ran if I'm only deploying one service?

`okteto deploy` is going to replace `okteto stack deploy` and since `okteto stack deploy` can have those arguments, `okteto deploy` should have those too if we want to remove `okteto stack deploy`, we can't remove it until if the functionality is not the same as it was.

> What if the deploy uses images built from the compose? For example the command could be echo ${OKTETO_BUILD_MYSQL_IMAGE} ${OKTETO_BUILD_FRONTEND_IMAGE}. Wouldn't we need to build all from the compose too?

In that case it shouldn't be covered by this PR. If that's the case, what the user probably needs to do is to add the `depends_on` field to the service specifying the other builds that should be triggered

Reply:
@jLopezbarb I've been checking the `InferFromStack` function to check if we can reuse that. Seems like we are modifying the `Stack` in there. I don't think we'll be able to reuse it

```go
...
		buildInfo := svcInfo.Build // Here svcInfo.Build is a pointer, so we are pointing to the same struct

		switch {
		case buildInfo != nil && len(svcInfo.VolumeMounts) > 0:
			if svcInfo.Image != "" {
				buildInfo.Image = svcInfo.Image // Here we are modifying the buildInfo, which under the hat modifies svcInfo.Build
			}
...
```


Another thing that concerns me is that we are unifying all the different okteto manifests under the god struct `model.Manifest`, yet we need to handle all cases every time we use it. This is a problem for extending the functionality, as it seems we cannot rely on the `Manifest` struct to tell the hole story. I think this problem is out of scope, but has great impact on the bugfixes/features we are working on

Reply:
We could create a function that checks which services has to be built from the stack manifest then. We can use that function on the deploy command (for now) until we refactor the manifest parsing and find a better way of doing it

Reply:
@jLopezbarb I've managed to do what you've mentioned `the ones from that the user selected on compose + the ones that are on the manifest`, and the current scenario works.

I've managed to get the services to be built by doing `(services to deploy ∩ services from stack) + services from manifest build section` 

I still need to add some tests, and I want to see if there's a cleaner way to do this without getting information from so many sources which could easily get out of sync and introduce bugs in the future